Add Path Planning Section to Behaviour NUbook Page (#175)
NUbots/NUbook: src/book/02-system/02-subsystems/06-behaviour.mdx

ScriptTuner also allows for "locking" and "unlocking" servos on the robot. When a servo is "locked", its position can only be changed in ScriptTuner. When it is "unlocked", the servo can be moved physically on the robot, which will update the position value in the script. Locking a servo involves emitting a ServoCommand for the current position of the servo. Unlocking the servo involves emitting a ServoCommand with zero torque and gain.

To learn how to use ScriptTuner, read the [ScriptTuner guide](/guides/main/tuning-and-running-scripts#script-tuner).

## Path planning

> [SimpleWalkPathPlanner](https://github.com/NUbots/NUbots/tree/main/module/behaviour/planning/SimpleWalkPathPlanner) is the current implemenation of path-planning for the robot and currently (as of **_09/2021_**) uses a combination of messages from vision and localisation in order to determine messages to emit which motion can then interpret to control the positions the robot moves to on the field during play.

Reviewer: ```suggestion
We use a [SimpleWalkPathPlanner](https://github.com/NUbots/NUbots/tree/main/module/behaviour/planning/SimpleWalkPathPlanner) to implement path-planning for the robot. The path planner uses a combination of messages from Vision and Localisation to determine which messages to emit. The motion subsystem can then interpret these messages to control the positions and locations the robot moves to on the field during play.
```
I'm not sure we need to specify a date here... It will probably get out of date at some point although including dates isn't generally the done thing

Reviewer 2: Yeah not sure on the date thing, probs best to remove but I like the concept. I was thinking Git already dates things so we know when it was updated. But it might be cool to have some way of seeing on the site itself when things have last been updated, maybe also who contributed so people know who to ask for more info. Maybe some sort of footer on the page.

Reviewer 2: I'd also recommend linking to the vision, localisation and motion pages where you mention the systems (if they haven't already been linked on the page).

Reviewer: That's an interesting idea @ysims are you thinking like a gitblame? We could look into that...

Reviewer 3: I've a similar idea of listing who contributed to a page since the beginning, primarily to encourage people to contribute by publicly giving credit. But knowing who to contact about a page is also a good reason to have that. And a last update date on each page would be good too 👍.

Reviewer: I've put it on the todo list, we can look into getting it done :D
https://github.com/NUbots/NUbook/issues/186

> [A state-machine](https://www.freecodecamp.org/news/state-machines-basics-of-computer-science-d42855debc66/) is currently being implemented to use a switch case so that depending on the latest motion command the path planner will call a function to perform a specific 'style' of path. For example, this may be following a [Bezier curve](https://javascript.info/bezier-curve#maths) to reach the required orientation and position to perform a kick toward the goal if the robot is in an offensive position or role which will be decided within Soccer Strategy, or if the robot is a goalie or similar defensive role it moves to the required position in a straight linear path to get there more quickly.

Reviewer: ```suggestion
We also use a [state-machine](https://www.freecodecamp.org/news/state-machines-basics-of-computer-science-d42855debc66/) to compare the latest motion command and call a function accordingly to perform a specific 'style' of path based on the style it is currently performing. For example if the robot is in an offensive position (a role determined by the Soccer Strategy module), the robot may be following a [Bezier curve](https://javascript.info/bezier-curve#maths) to reach the required orientation and position to perform a kick toward the goal. Alternatively if the robot is a goalie or similar defensive role, it will move to the required position in a linear path in order to get there more quickly.
```

> [A state-machine](https://www.freecodecamp.org/news/state-machines-basics-of-computer-science-d42855debc66/) is currently being implemented to use a switch case so that depending on the latest motion command the path planner will call a function to perform a specific 'style' of path. For example, this may be following a [Bezier curve](https://javascript.info/bezier-curve#maths) to reach the required orientation and position to perform a kick toward the goal if the robot is in an offensive position or role which will be decided within Soccer Stratergy, or if the robot is a goalie or similar defensive role it moves to the required position in a straight linear path to get there more quickly.

Reviewer 2: Probably change this assuming that PR has been merged and merge these two PRs around the same time, otherwise you'll need a follow up PR to adjust this wording.

Reviewer 2: ```suggestion
[A state-machine](https://www.freecodecamp.org/news/state-machines-basics-of-computer-science-d42855debc66/) is currently being implemented to use a switch case so that depending on the latest motion command the path planner will call a function to perform a specific 'style' of path. For example, this may be following a [Bezier curve](https://javascript.info/bezier-curve#maths) to reach the required orientation and position to perform a kick toward the goal if the robot is in an offensive position or role which will be decided within Soccer Strategy, or if the robot is a goalie or similar defensive role it moves to the required position in a straight linear path to get there more quickly.
```

> ### Implemenation of Bezier Curves

Reviewer 2: ```suggestion
### Implementation of Bezier Curves
```

> ### Implemenation of Bezier Curves

Reviewer 2: General stuff about Bezier curves can go in the maths page. Specifics of how it's use for the walk path planning can go in here.

> Bezier curves are functions that are defined by a limited series of control points. Based on emits from localisation and vision modules to determine the positions of the robot, ball, and goals, these control points, and as such the curve the robot would follow, is dynamically corrected in real-time.

Reviewer: ```suggestion
Bezier curves are functions that are defined by a limited series of control points. Based on emits from Localisation and Vision modules which determine the positions of the robot, ball, and goals, these control points, and as such the curve the robot would follow, is dynamically corrected in real-time.
```
